galley/pkg/config/processing/snapshotter/analyzingdistributor.go

```diff
@@ -140,22 +157,33 @@ func (d *AnalyzingDistributor) analyzeAndDistribute(cancelCh chan struct{}, name
 	d.s.Analyzer.Analyze(ctx)
 	scope.Analysis.Debugf("Finished analyzing the current snapshot, found messages: %v", ctx.messages)
 
-	// Only keep messages for resources in namespaces we want to analyze if the
-	// message doesn't have an origin (meaning we can't determine the
-	// namespace). Also kept are cluster-level resources where the namespace is
-	// the empty string. If no such limit is specified, keep them all.
 	var msgs diag.Messages
-	if len(namespaces) == 0 {
-		msgs = ctx.messages
-	} else {
-		for _, m := range ctx.messages {
-			if m.Origin != nil && m.Origin.Namespace() != "" {
-				if _, ok := namespaces[m.Origin.Namespace()]; !ok {
-					continue
-				}
+FilterMessages:
+	for _, m := range ctx.messages {
+		// Only keep messages for resources in namespaces we want to analyze if the
+		// message doesn't have an origin (meaning we can't determine the
+		// namespace). Also kept are cluster-level resources where the namespace is
+		// the empty string. If no such limit is specified, keep them all.
+		if len(namespaces) > 0 && m.Origin != nil && m.Origin.Namespace() != "" {
+			if _, ok := namespaces[m.Origin.Namespace()]; !ok {
+				continue FilterMessages
 			}
-			msgs = append(msgs, m)
 		}
+
+		// Filter out any messages that match our suppressions.
+		for _, s := range d.s.Suppressions {
+			if m.Origin == nil || s.Code != m.Type.Code() {
+				continue
+			}
+
+			if match, err := filepath.Match(s.ResourceName, m.Origin.FriendlyName()); err != nil || !match {
```

Comment:
This is potentially confusing and deserves at least an explanatory comment. Someone will think files are involved here, when we're actually reusing file matching logic for a non-file use case. 

The downside of the filepath matcher is that we're well beyond basic globbing. TBH it's probably overkill, and I vaguely worry all the "extra" stuff we're implicitly getting could lead to confusing behavior down the line. 

Digging around a bit, I found https://github.com/ryanuber/go-glob, which does exactly what we want: `*` as wildcard and no more. It looks like we already pull it in indirectly (I see it in our `go.mod` file). Is it worth upgrading this to a direct dependency so we can use it?  

If we do stick with the filepath matcher, let's document clearly why we've done so. Our user-facing documentation should probably not mention more functionality than the wildcards.

Reply:
This is a good suggestion. Normally I'm really worried about adding new dependencies, but given we already indirectly depend on this package, and it appears to be relatively stable/small, I went ahead and updated the code to directly use it.